### ISSUE TITLE
yarn install fails when package.json contains recently published packages

### ISSUE THREAD

**Issue by rarkins** [yarnpkg/yarn#5169]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

If a package.json contains a dependency with a recently-published dependency, "npm install" works but "yarn install" fails, reporting 404 for that particular version.

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Create a package.json that depends on package "a" you control.
2. Publish a new version of "a"
3. Update package.json to include the exact version of "a" you just published
4. Run yarn install

What happens: yarn install fails

**What is the expected behavior?**

yarn install should not fail, it should proxy requests to npmjs if necessary.

**Please mention your node.js, yarn and operating system version.**

Node.js 8.x, yarn 1.3.2, Ubuntu 16.04

** My question **

Is it OK if I just force yarn to always use the npmjs registry instead of the yarn proxy? e.g. an `.npmrc` containing `registry=https://registry.npmjs.org/`. The benefits of the yarn registry are non-existent or unknown, the downsides are evident.

**Comment by rally25rs** [comment]
> Is it OK if I just force yarn to always use the npmjs registry instead of the yarn proxy? e.g. an .npmrc containing registry=https://registry.npmjs.org/

Yes that would work fine. The NPM registry was having some seemingly random 404 issues yesterday too according tot heir status page; not sure if that is related.

**Comment by rarkins** [comment]
@rally25rs no, this has happened for a long time and unrelated from yesterday’s problem. I run a tool that detects new dependency versions and updates package and lock files, so I see this a lot.

**Comment by rally25rs** [comment]
Interesting. As far as I know the yarn registry should check it's own cache of known packages, and if not found, hit NPM. I'm not entirely sure how it works though. Maybe if that package already existed, the Yarn registry uses the already known/cached metadata?

@BYK any idea why would best handle this? (unassigning myself)

**Comment by rarkins** [comment]
@rally25rs I had hoped that was the case too, but when investigating failed lock file causes I found 100% of failures were yarn users and it was always the dependency being updated that `yarn install` reported as 404’ing. Doesn’t ever happen on npm install`.

**Comment by rarkins** [comment]
This issue has not reoccurred since when I observed it pre-Christmas, so I think it can be closed.

**Comment by BYK** [comment]
I'll double check our CDN config and see if there's anything wrong there. Thanks for bringing this up @rarkins .

**Comment by rarkins** [comment]
@BYK I recently made changes to the app's backend so that it will generate PRs to downstream packages as fast as possible once changes to upstream ones are detected. Previously this was only checked on an hourly cycle. Anyway, if the problem was still present then this speeding up would definitely have triggered it. Now, most `yarn installs` would be running within seconds of `npmjs` notifying us of a package update, so that's why I felt pretty confident there is no problem. I've also added some log metrics and alerts in case it ever does come back again, so I should be able to alert you quickly if that ever happens.
